src/Microsoft.CorrelationVector/CorrelationVectorV1.cs

```diff
@@ -0,0 +1,404 @@
+                {
+                    return this.Value;
+                }
+                next = snapshot + 1;
+                if (CorrelationVectorV1.IsOversized(this.BaseVector, next, this.Version))
+                {
+                    this.immutable = true;
+                    return this.Value;
+                }
+            }
+            while (snapshot != Interlocked.CompareExchange(ref this.extension, next, snapshot));
+            return string.Concat(this.BaseVector, ".", next);
+        }
+
+        public override string Base
+        {
+            get
+            {
+                int firstDotLocation = Value.IndexOf('.');
+                return Value.Substring(0, firstDotLocation);
+            }
+        }
+
+        public override int Extension
+        {
+            get
+            {
+                return this.extension;
+            }
+        }
+
+        /// <summary>
+        /// Returns a string that represents the current object.
+        /// </summary>
+        /// <returns>A string that represents the current object.</returns>
+        public override string ToString()
+        {
+            return this.Value;
+        }
+
+        /// <summary>
+        /// Determines whether two instances of the <see cref="CorrelationVectorV1"/> class
+        /// are equal. 
+        /// </summary>
+        /// <param name="vector">
+        /// The correlation vector you want to compare with the current correlation vector.
+        /// </param>
+        /// <returns>
+        /// True if the specified correlation vector is equal to the current correlation
+        /// vector; otherwise, false.
+        /// </returns>
+        public bool Equals(CorrelationVectorV1 vector)
+        {
+            return string.Equals(this.Value, vector.Value, StringComparison.Ordinal);
+        }
+
+        private CorrelationVectorV1(string baseVector, int extension, CorrelationVectorVersion version, bool immutable)
+        {
+            this.BaseVector = baseVector;
+            this.extension = extension;
+            this.Version = version;
+            this.immutable = immutable || CorrelationVectorV1.IsOversized(baseVector, extension, version);
+        }
+
+        private static string GetUniqueValue(CorrelationVectorVersion version)
+        {
+            if (CorrelationVectorVersion.V1 == version)
+            {
+                byte[] bytes = Guid.NewGuid().ToByteArray();
+                return Convert.ToBase64String(bytes, 0, 12);
+            }
+            else if (CorrelationVectorVersion.V2 == version)
+            {
+                return Guid.NewGuid().GetBaseFromGuid();
+            }
+            else
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, "Unsupported correlation vector version: {0}", version));
+            }
+        }
+
+        private static CorrelationVectorVersion InferVersion(string correlationVector, bool reportErrors)
```

Comment:
There shouldnt be a need to infer version in each of the version implementations